.github/actions/build_documentation/documentation/s3.rb

```diff
@@ -0,0 +1,114 @@
+module Documentation
+  class S3
+    attr_reader :client, :bucket
+
+    def initialize(config)
+      @bucket = config.delete(:bucket)
+      @client = Aws::S3::Client.new(config)
+    end
+
+    def upload!(version, source_dir)
+      # Upload new documentation to _tmp
+      each_file(source_dir) do |file_path|
+        put(File.join('_tmp', file_path.sub("#{source_dir}/", '')), file_path)
+      end
+
+      # # Delete existing documentation for this version
```

Comment:
I assume this will get uncommented out for final merge?

Reply:
yes, sorry, this isnt ready for final review yet.